diff --git a/vendor/magento/module-cms/Block/Adminhtml/Page/Widget/Chooser.php b/vendor/magento/module-cms/Block/Adminhtml/Page/Widget/Chooser.php
index 32d36a1869..140128ae4b 100644
--- a/vendor/magento/module-cms/Block/Adminhtml/Page/Widget/Chooser.php
+++ b/vendor/magento/module-cms/Block/Adminhtml/Page/Widget/Chooser.php
@@ -113,7 +113,7 @@ class Chooser extends \Magento\Backend\Block\Widget\Grid\Extended
      */
     public function getRowClickCallback()
     {
-        $chooserJsObject = $this->getId();
+        $chooserJsObject = $this->_escaper->escapeJs($this->getId());
         $js = '
             function (grid, event) {
                 var trElement = Event.findElement(event, "tr");
